### PR TITLE
[bitnami/kube-prometheus] Added `routePrefix` and `portName` to `alertmanager` and `prometheus`

### LINKED ISSUE #5571: Request to support the `portName` field on both AlertManager and Prometheus
**Which chart**:
Name: kube-prometheus
Version: 2.2.1

**Is your feature request related to a problem? Please describe.**
The chart currently will create both `Prometheus` and `AlertManager` with the default `portName`, which defaults to `web`, resulting in two kubernetes service objects using the same port but different port name (**for example, a `ClusterIP` service with `TCP/9090 = http` and a `Headless` service with the same port called `web`**). This is causing issue with the Istio Envoy sidecar. I have manually patch the `Prometheus` and `AlertManager` object afterward to match the port name and problem is resolved.

**Describe the solution you'd like**
Addition of `alertmanager.portName` and `prometheus.portName`

**Describe alternatives you've considered**
Manual patching the objects after deployment. Not elegant but it works for now.

### LINKED ISSUE #5572: Request to support the `routePrefix` field on AlertManager
**Which chart**:
Name: kube-prometheus
Version: 2.2.1

**Is your feature request related to a problem? Please describe.**
I was trying to expose `AlertManager` via an Istio ingress gateway using a sub path, but I notice `alertmanager.routePrefix` isn't supported today. Even though it is a supported option for `Prometheus`.

**Describe the solution you'd like**
Addition of `alertmanager.routePrefix`

**Describe alternatives you've considered**
Manually patching the object after deployment.

### PR DESCRIPTION
**Pull request by petersin0422**
**Description of the change**

Adding support for the following values:
- prometheus.portName
- alertmanager.portName
- alertmanager.routePrefix

**Benefits**

Without changing the `portName` the default value of `web` will be used, resulting in two Kubernetes service objects (one `ClusterIP` and one `Headless`) pointing to the same port but with different names.

services

This doesn't work well with the Istio Envoy sidecar. The above will results in a `404 NR` error when directly accessing the pod. Which is what Prometheus is going to do when scraping their metrics.

istio-log

The `alertmanager.routePrefix` is to enable AlertManager to serve their website in a sub path, which is what we are doing with the Istio ingress gateway.

**Possible drawbacks**

None.

**Applicable issues**

  - fixes #5572
  - fixes #5571 

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/kube-prometheus/Chart.yaml [hunk 1]
```diff
@@ -34,4 +34,4 @@ sources:
   - https://github.com/bitnami/bitnami-docker-prometheus
   - https://github.com/bitnami/bitnami-docker-alertmanager
   - https://github.com/prometheus-operator/kube-prometheus
-version: 4.0.1
+version: 4.0.2
```
Comment (Mauraza) [inline bitnami/kube-prometheus/Chart.yaml:37]: as it is a feature, could you increase the version to 4.1.0?

**Review by Mauraza — COMMENTED** [review]
Hi @pankajkoti,

Thanks a lot for the contribution!!, could you check the small comment?

**Comment by pankajkoti** [comment]
Hi @Mauraza, I guess I am wrongly tagged here. Perhaps you meant to tag @petersin0422

File: bitnami/kube-prometheus/Chart.yaml (hunk 1 above)
Comment (petersin0422) [reply bitnami/kube-prometheus/Chart.yaml]: Done! 😄

**Comment by Mauraza** [comment]
Hi @petersin0422,

LGTM. Thanks again for the contribution!

**Pull request merged**